--- a/src/uu/mv/src/mv.rs	2026-06-27 23:42:40.973211345 +0800
+++ b/src/uu/mv/src/mv.rs	2026-06-27 23:42:40.982990875 +0800
@@ -111,7 +111,7 @@
 impl Default for Options {
     fn default() -> Self {
         Self {
-            overwrite: OverwriteMode::default(),
+            overwrite: OverwriteMode::Force,
             backup: BackupMode::default(),
             suffix: backup_control::DEFAULT_BACKUP_SUFFIX.to_owned(),
             update: UpdateMode::default(),
@@ -347,7 +347,7 @@
     } else if matches.get_flag(OPT_FORCE) {
         OverwriteMode::Force
     } else {
-        OverwriteMode::Default
+        OverwriteMode::Force
     }
 }
 
